test/test/lock/LockTests.java

```diff
@@ -10,6 +10,34 @@
 import one.profiler.test.TestProcess;
 
 public class LockTests {
+    private static void contendedLocks(TestProcess p, int threshold, double minRatio) throws Exception {
+        Output out = p.profile("-e lock --lock " + threshold + " --threads -o collapsed");
+        Output waitingLocks = out.filter("LockProfiling\\.contend;java\\.lang\\.Object");
+
+        double ratio = 100 * out.ratio("contend-0.5-", "contend-0.05-");
+
+        System.out.println("threshold = " + threshold + ", ratio = " + ratio + ", minRatio = " + minRatio);
+
+        assert (ratio >= minRatio) || (Double.isNaN(ratio) && Double.isNaN(minRatio));
+    }
+
+    // 0 is equivalent to disabling sampling of locks, so all profiles are included.
+    @Test(mainClass = LockProfiling.class, inputs = { "0", "75" })
+    @Test(mainClass = LockProfiling.class, inputs = { "1", "75" })
+    @Test(mainClass = LockProfiling.class, inputs = { "10000", "75" })
+
+    // Large (for the specific paylod) interval value skews the sampled lock
+    // contention distribution.
```

Comment:
Is it an expected behavior? Is it what we want to assert here?

Reply:
This is an expected behavior. As the `interval` increases, lock contentions with longer durations will be favored.